ui/src/data_explorer/reducers/ui.js

```diff
@@ -0,0 +1,27 @@
+const initialState = {
+  queryIDs: [],
+};
+
+export default function ui(state = initialState, action) {
+  switch (action.type) {
+    case 'ADD_QUERY': {
+      const {queryID} = action.payload;
+      const newState = {
+        queryIDs: state.queryIDs.concat(queryID),
+      };
+
+      return {...state, ...newState};
+    }
+
+    case 'DELETE_QUERY': {
+      const {queryID} = action.payload;
+      const newState = {
+        queryIDs: state.queryIDs.filter(id => id !== queryID),
+      };
+
+      return {...state, ...newState};
```

Comment:
Does line 22 do anything? Am I wrong in thinking this should be simply newState, or is there something I'm missing?